### PR TITLE
fix(Badges)MBA-43:fix search query -> now open refused badges

### PR DESCRIPTION
**Pull request by Rseuret**
## Describe your changes
open refused badges if elems match search
## Issue ticket number and link
https://jira.support-ent.fr/browse/MBA-43
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts [hunk 1]
```diff
@@ -43,9 +46,13 @@ class Controller implements ng.IController, IViewModel {
 
     $onInit() {
         if (this.$scope.vm.isDefaultOpenedList) this.isOpenedList = this.$scope.vm.isDefaultOpenedList;
+          this.isOpenedOption =  (this.$scope.vm.isOpenedOption !== undefined) ? this.$scope.vm.isOpenedOption : false;
```
Comment (BastienBenard) [inline src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts:49]: Est ce que tu ne peux pas faire directement `this.isOpenedOption = !!this.$scope.vm.isOpenedOption` ?

File: src/main/resources/public/ts/controllers/minibadge.controller.ts [hunk 2]
```diff
@@ -65,6 +65,7 @@ class Controller implements ng.IController, ViewModel {
                     this.refusedBadges = this.badges ? this.badges.filter((badge: Badge) => badge.isRefused()) : [];
                 }
 
+                this.isOpenedOption = this.payload.query && this.payload.query.length && this.payload.query.length > 0;
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/controllers/minibadge.controller.ts:68]: this.payload.query && !!this.payload.query.trim().length
c'est suffisant

File: src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts [hunk 3]
```diff
@@ -43,9 +46,13 @@ class Controller implements ng.IController, IViewModel {
...
         this.$scope.$watch((): Badge[] => this.$scope.vm.badges, (newBadges: Badge[], oldBadges: Badge[]): void => {
             this.checkOpenedListStatus(newBadges, oldBadges);
         });
+        this.$scope.$watch(():boolean => this.$scope.vm.isOpenedOption, (newOption:boolean): void => {
+            this.isOpenedOption = newOption;
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts:54]: pas nécessaire avec un 2 way binding => 
scope : {
...
"paramètre": "="
...
}

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts [hunk 4]
```diff
@@ -43,6 +46,7 @@ class Controller implements ng.IController, IViewModel {
 
     $onInit() {
         if (this.$scope.vm.isDefaultOpenedList) this.isOpenedList = this.$scope.vm.isDefaultOpenedList;
+            this.isOpenedOption = !!this.$scope.vm.isOpenedOption
```
Comment (sjollois) [inline src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts:49]: indentation

**Review by sjollois — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

**Review by sjollois — APPROVED** [review]

File: src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts [hunk 5]
```diff
@@ -16,13 +16,15 @@ interface IViewModel {
     CARD_FOOTER: typeof CARD_FOOTER;
     lang: typeof lang;
     isOpenedList: boolean;
+    isOpenedOption:boolean;
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts:19]: Je vais faire le tatillon, et je le fais avant que ce soit quelqu'un d'autre qui le fasse 😛 
donc un petit coup de ALT + MAJ + L pour les espaces serais le bienvenu 😄

File: src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts [hunk 6]
```diff
@@ -43,6 +46,7 @@ class Controller implements ng.IController, IViewModel {
 
     $onInit() {
         if (this.$scope.vm.isDefaultOpenedList) this.isOpenedList = this.$scope.vm.isDefaultOpenedList;
+        this.isOpenedOption = !!this.$scope.vm.isOpenedOption
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts:49]: ;

File: src/main/resources/public/ts/controllers/minibadge.controller.ts [hunk 7]
```diff
@@ -66,6 +66,7 @@ class Controller implements ng.IController, ViewModel {
                     this.refusedBadges = this.badges ? this.badges.filter((badge: Badge) => badge.isRefused()) : [];
                 }
 
+                this.isOpenedOption = this.payload.query && !!this.payload.query.trim().length
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/controllers/minibadge.controller.ts:69]: ;

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

**Review by qperie — APPROVED** [review]

File: src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts (hunk 5 above)
Comment (qperie) [reply src/main/resources/public/ts/directives/badge-list/badge-list.directive.ts:19]: Je vais tatillonner aussi : @qpic-cgi on a pas tous les mêmes raccourcis donc ALT+MAJ+L marche pas forcément pour Rémy x)

**Pull request merged**